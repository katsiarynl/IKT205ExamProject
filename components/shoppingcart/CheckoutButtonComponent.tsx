import { useNavigation } from "@react-navigation/native";
import React, { useContext } from "react";
import { Pressable, Text, View } from "react-native";
import { RestaurantContext } from "../../context";
import cartstyle from "../../styles/cartstyles";
import POSTStripe from "../../utilities/POSTStripe";
import { UserContext } from "../Auth/userContext";
import { NavigationSignIn } from "../../types/navigationTypes";
import { checkIfAddressExists } from "../../utilities/checkAddress";

async function NavitagateToStripe(callback, navigate, ordered_dishes) {
  const link = await callback;

  navigate("stripe", { link, ordered_dishes });
  return link;
}
export default function CheckoutButtonComponent() {
  const navigator = useNavigation();
  const NavigatorHome = useNavigation<NavigationSignIn>();
  const { state } = useContext(RestaurantContext);

  const { isloggedIn } = useContext(UserContext);

  const handleCheout = async () => {
    if (isloggedIn) {
      const addressExists = await checkIfAddressExists();
      if (addressExists) {
        await NavitagateToStripe(
          POSTStripe(state.cartItems),
          navigator.navigate,
          state.cartItems
        );
      } else {
        NavigatorHome.navigate("AddressForm");
      }
    } else {
      NavigatorHome.navigate("SignIn");
    }
  };

  return (
    <View
      style={{
        flex: 2,
        justifyContent: "center",
        alignItems: "center",
      }}
    >
      <Pressable
        style={({ pressed }) => [
          cartstyle.checkoutbutton,
          state.cartItems.length !== 0
            ? { backgroundColor: "#F2C94C", ...(pressed && { opacity: 0.4 }) }
            : { backgroundColor: "gray" },
        ]}
        disabled={state.cartItems.length === 0}
        onPress={handleCheout}
        testID="CheckoutNavigate"
      >
        <View style={{ backgroundColor: "white " }}>
          <Text
            style={{
              fontSize: 25,
              fontWeight: "bold",
            }}
          >
            Checkout
          </Text>
        </View>
      </Pressable>
    </View>
  );
}
