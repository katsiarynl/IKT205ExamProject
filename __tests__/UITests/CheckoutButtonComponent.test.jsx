import React from "react";
import { create } from "react-test-renderer";
import CheckoutButtonComponent from "../../components/shoppingcart/CheckoutButtonComponent";
const mockedNavigate = jest.fn();
//https://stackoverflow.com/questions/61781274/how-to-mock-usenavigation-hook-in-react-navigation-5-0-for-jest-test
jest.mock("@react-navigation/native", () => ({
  useNavigation: () => ({ navigate: mockedNavigate }),
}));
const tree = create(<CheckoutButtonComponent />);

test("snapshot", () => {
  expect(tree).toMatchSnapshot();
});
