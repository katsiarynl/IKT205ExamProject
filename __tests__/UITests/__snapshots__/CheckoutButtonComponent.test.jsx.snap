// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for CheckoutButtonComponent 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 2,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 5,
          "height": "65%",
          "justifyContent": "center",
          "width": "92%",
        },
        {
          "backgroundColor": "gray",
        },
      ]
    }
    testID="CheckoutNavigate"
  >
    <View
      style={
        {
          "backgroundColor": "white ",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        Checkout
      </Text>
    </View>
  </View>
</View>
`;
